packages/server/src/schema/uuid/abstract-entity/utils.ts

```diff
@@ -59,3 +74,288 @@ export function createEntityResolvers<
+  })
+
+  await assertUserIsAuthorized({
+    userId,
+    dataSources,
+    message: 'You are not allowed to add revision to this entity.',
+    guard: serloAuth.Uuid.create('EntityRevision')(scope),
+  })
+
+  const entity = await dataSources.model.serlo.getUuidWithCustomDecoder({
+    id: entityId,
+    decoder: childDecoder,
+  })
+
+  let isAutoreviewEntity = false
+
+  if (entity.taxonomyTermIds) {
+    isAutoreviewEntity = await verifyAutoreviewEntity(
+      entity.taxonomyTermIds,
+      dataSources
+    )
+  }
+
+  if (entity.parentId && parentDecoder) {
+    const parent = await dataSources.model.serlo.getUuidWithCustomDecoder({
+      id: entity.parentId,
+      decoder: parentDecoder,
+    })
+
+    if (parent.taxonomyTermIds) {
+      isAutoreviewEntity = await verifyAutoreviewEntity(
+        parent.taxonomyTermIds,
+        dataSources
+      )
+    }
+  }
+
+  if (!isAutoreviewEntity && !needsReview) {
+    await assertUserIsAuthorized({
+      userId,
+      dataSources,
+      message: 'You are not allowed to skip the reviewing process.',
+      guard: serloAuth.Entity.checkoutRevision(scope),
+    })
+  }
+
+  const fields = removeUndefinedFields(
+    inputFields as { [key: string]: string | undefined }
+  )
+
+  const inputPayload = {
+    changes,
+    entityId,
+    needsReview: isAutoreviewEntity ? false : needsReview,
+    subscribeThis,
+    subscribeThisByEmail,
+    fields,
+  }
+  const { success, revisionId } =
+    await dataSources.model.serlo.addEntityRevision({
+      revisionType,
+      userId,
+      input: inputPayload,
+    })
+
+  return {
+    revisionId,
+    success,
+    query: {},
+  }
+}
+
+export async function getEntity<S extends Model<'AbstractEntity'>>(
+  entityId: number,
+  dataSources: Context['dataSources'],
+  decoder: t.Type<S, unknown>
+) {
+  return await dataSources.model.serlo.getUuidWithCustomDecoder({
+    id: entityId,
+    decoder,
+  })
+}
+
+export async function verifyAutoreviewEntity(
+  taxonomyTermIds: number[],
+  dataSources: Context['dataSources']
+): Promise<boolean> {
+  return (
+    await Promise.all(
+      taxonomyTermIds.map((id) => checkAnyParentAutoreview(id, dataSources))
+    )
+  ).every((x) => x)
+}
+
+async function checkAnyParentAutoreview(
+  taxonomyTermId: number,
+  dataSources: Context['dataSources']
+): Promise<boolean> {
+  if (autoreviewTaxonomyIds.includes(taxonomyTermId)) return true
+
+  const taxonomyTerm = await dataSources.model.serlo.getUuidWithCustomDecoder({
+    id: taxonomyTermId,
+    decoder: TaxonomyTermDecoder,
+  })
+
+  return (
+    taxonomyTerm.parentId !== null &&
+    (await checkAnyParentAutoreview(taxonomyTerm.parentId, dataSources))
+  )
+}
+
+function removeUndefinedFields(inputFields: {
+  [key: string]: string | undefined
+}) {
+  const fields: {
+    [key: string]: string
+  } = {}
+
+  for (const [key, value] of Object.entries(inputFields)) {
+    if (value) {
+      fields[key] = value
+    }
+  }
+
+  return fields
```

Comment:
Try:

```suggestion
  return R.filter(value => value != undefined, inputFields)
```

See https://ramdajs.com/repl/?v=0.28.0#?R.filter%28value%20%3D%3E%20value%20%21%3D%20null%2C%20%7B%20a%3A%20undefined%2C%20b%3A%202%7D%29 (Hopefully the types work fine)